src/Zynga/Framework/PgData/V1/Testing/Mock/PgModel.hh

```diff
@@ -0,0 +1,46 @@
+<?hh //strict
+
+namespace Zynga\Framework\PgData\V1\Testing\Mock;
+
+use Zynga\Framework\PgData\V1\Interfaces\PgModel\ReaderInterface;
+use Zynga\Framework\PgData\V1\Interfaces\PgModel\StatsInterface;
+use Zynga\Framework\PgData\V1\Interfaces\PgModel\WriterInterface;
+use Zynga\Framework\PgData\V1\Interfaces\PgRowInterface;
+use Zynga\Framework\PgData\V1\Testing\Mock\PgReaderWriter;
+use Zynga\Framework\PgData\V1\PgModel as Base;
+
+class PgModel extends Base {
+  private PgReaderWriter $_readerWriter;
+
+  public function __construct() {
+    $this->_readerWriter = new PgReaderWriter();
+  }
+
+  public function getMockReaderWriter(): PgReaderWriter {
+    return $this->_readerWriter;
+  }
+
+  public function getDataCacheName(): string {
+    return 'Mock\PgData';
+  }
+
+  public function getResultSetCacheName(): string {
+    return 'Mock\PgResultSet';
+  }
+
+  public function getReadDatabaseName(): string {
```

Comment:
When you are using PgReaderWriter its not going to use Mock database right? Maybe i am misunderstanding the use of readerWriter with Mock read and write database

Reply:
That is correct. When using this Mock class, it will only use the the PgReaderWriter,  but still need to define these methods. Since these methods should not be called, we could change the name to be something like "Should Not Be Called" to make it clearer that we are not using the mock databases.

Reply:
What if you just return '' instead of 'mock' and add a comment that this model uses PgReaderWriter for the DB. Personally i got confused coz return 'mock' made me think its going to the mock db driver